components/screens/ledger-connect/LedgerConnectForSign.tsx

```diff
@@ -0,0 +1,150 @@
+import useLedgerTransport from "@/hooks/useLedgerTransport";
+import { useNavigate, useSearchParams } from "react-router-dom";
+import { useEffect, useRef } from "react";
+import Header from "@/components/GeneralHeader";
+import ledgerConnectingImage from "@/assets/images/ledger-connecting.png";
+import { twMerge } from "tailwind-merge";
+
+export default function LedgerConnectForSign() {
+  const { transport, connect, isAppOpen, isConnecting } = useLedgerTransport();
+  const [searchParams] = useSearchParams();
+  const navigate = useNavigate();
+  const redirect = searchParams.get("redirect");
+  const state = searchParams.get("state");
+  const calledConnectOnce = useRef(false);
+  const callCheckNeedRetryOnce = useRef(false);
+  const [showRetry, setShowRetry] = useState(false);
+
+  const redirectBack = useCallback(() => {
+    if (!redirect) {
+      return;
+    }
+
+    navigate(
+      {
+        pathname: redirect,
+      },
+      {
+        state: JSON.parse(decodeURIComponent(state ?? "") ?? "{}"),
+      },
+    );
+  }, [redirect, state]);
+
+  const connectDevice = async () => {
+    // Do nothing if the connection is in progress
+    if (isConnecting || !redirect) {
+      return;
+    }
+
+    try {
+      await connect();
+    } catch (error) {
+      setTimeout(
+        () =>
+          navigate({
+            pathname: "/ledger-connect-for-sign-failed",
+            search: `?redirect=${redirect}&state=${state}`,
+          }),
+        1000, // Delay to prevent the page from flickering
+      );
+    }
+  };
+
+  const retry = () => {
+    setShowRetry(false);
+    connectDevice();
+    setTimeout(() => {
+      setShowRetry(true);
+    }, 1000); // Show retry button after 1 seconds
+  };
+
+  useEffect(() => {
+    if (!calledConnectOnce.current) {
```

Comment:
Maybe let's do it like we did elsewhere, just for consistency:

```typescript
    if (calledOnce.current) return;
    calledOnce.current = true;
```

Reply:
https://github.com/forbole/kastle/pull/41/commits/7be98d55211ae48a9bed0354ce056c87dabc1860